### ISSUE TITLE
NextCloud Integration

### ISSUE THREAD

**Issue by mmosley77** [ondras/my-mind#155]
Would it be possible to make integration with NextCloud happen? There is another mind mapping app setup to do this found here https://github.com/ACTom/files_mindmap but it would be great if users that like the My Mind app better to do the same.

**Comment by joho1968** [comment]
Nextcloud is an open and online "hub" environment for apps, file storage, contacts, etc that can be self-hosted.

https://docs.nextcloud.com/server/latest/developer_manual/

**Comment by benchonaut** [comment]
well it seems to work "somehow" using 

`/remote.php/files`

 (
 * my-mind cant handle 201
 * but saving/ loading works 
 *  though the files apear under a different user than the one currently in the web session 
 
 ---
 
![grafik](https://user-images.githubusercontent.com/12718580/200429195-614fbba7-3251-457a-bc3e-8c8957f62fa0.png)
 
 )

**Comment by melvincarvalho** [comment]
HTTP 201 should be fixed in: https://github.com/ondras/my-mind/issues/171

It will need to run through Make before it gets into the main my-mind.js

I've been testing with my own DAV server locally and it works well, 201 doesnt give an alert.  I believe NextCloud has WebDAV too, so this could be a great back end.

**Comment by ondras** [comment]
> It will need to run through Make before it gets into the main my-mind.js

Done, and synced to the github-hosted demoversion.

**Comment by joho1968** [comment]
On https://my-mind.github.io, if I enter "Generic WebDAV" as the storage target, and click on "Save" without entering an URL, I get this in a modal:

```
<html>
<head><title>405 Not Allowed</title></head>
<body bgcolor="white">
<center><h1>405 Not Allowed</h1></center>
</body>
</html>
```

**Comment by ondras** [comment]
> On https://my-mind.github.io, if I enter "Generic WebDAV" as the storage target, and click on "Save" without entering an URL, I get this in a modal:

...because the demo version at GitHub is hosted via GitHub Pages; this services does not provide any WebDAV endpoints.

**Comment by joho1968** [comment]
> @joho1968 you need to save your file somewhere. That will be a URL. Without it you get an error. If you have loaded the mindmap from a URL it should populate the box.

Yes, I understand that. Sorry, I should have been more clear in my "report". What I meant to say is that I don't think the button should do anything if the URL field is blank.

**Comment by ondras** [comment]
> What I meant to say is that I don't think the button should do anything if the URL field is blank.

Right, then perhaps you are reporting an unrelated issue "disable the button while the input is empty".

**Comment by benchonaut** [comment]
one last hint:
`409 Conflict` ( see above screenshot) should be handled , either by "first get then diff then push" or by "open dialog , let user choose to overwrite/rename/whatever"
